Automattic/wp-calypso
# Jetpack Settings: Add conditional messaging on credentials settings screen for backups and scan

Resolves https://github.com/Automattic/jetpack-backup-team/issues/322

Currently, on the credentials settings screen, we display a generic message informing the user that they can add credentials to enable restores. However, this message does not address the fact that credentials are also necessary for fixing threats. The goal of this PR is to enhance the messaging by considering the specific products associated with the site. This update will ensure that the displayed message is more relevant and informative, based on whether the site has backup, scan features, or both.

## Proposed Changes
* Add `useCredentialsIntroMessage` to add conditional messaging.
* If `jetpack/backup-restore-preflight-checks` feature flag is not enabled, we will display the existing generic message. This is because we want to roll out the new messages when we ship preflight checks.

| Site plan | Screenshot |
|---|---|
| Backup |  <img src="https://github.com/Automattic/wp-calypso/assets/1488641/53abdf02-34fa-4c5b-9c57-e8f21e7918ef" width="350" /> |
| Scan | <img src="https://github.com/Automattic/wp-calypso/assets/1488641/dbbe9417-1e7c-4564-95ff-ae39ea979d91" width="350" /> |
| Backup + Scan | <img src="https://github.com/Automattic/wp-calypso/assets/1488641/a3483ee2-5f28-4b2f-98f8-c00ce6d148b5" width="350" /> |

## Testing Instructions
Note: you need to append `?flags=jetpack/backup-restore-preflight-checks` to the Calypso Live or Jetpack Cloud Live to enable the feature flag that will let you see the new wording.

* Prepare self-hosted site with each scenario described before and connect them to Jetpack
  * 1 site with Jetpack VaultPress Backup plan
  * 1 site with Jetpack Scan plan
  * 1 site with Jetpack VaultPress Backup + Jetpack Scan plans
* Navigate to Jetpack > Settings in Calypso Blue or Jetpack Cloud
* Ensure you see the appropriate message in the yellow notice, as shared above.

## Pre-merge Checklist
